server/lyft/gateway/events_controller.go

```diff
@@ -1,289 +1,94 @@
 package gateway
 
 import (
-	"bytes"
-	"fmt"
-	"io/ioutil"
 	"net/http"
 
-	"github.com/google/go-github/v31/github"
-	"github.com/pkg/errors"
 	events_controllers "github.com/runatlantis/atlantis/server/controllers/events"
+	"github.com/runatlantis/atlantis/server/controllers/events/handlers"
 	"github.com/runatlantis/atlantis/server/events"
 	"github.com/runatlantis/atlantis/server/events/models"
 	"github.com/runatlantis/atlantis/server/events/vcs"
 	"github.com/runatlantis/atlantis/server/logging"
-	"github.com/runatlantis/atlantis/server/lyft/aws/sns"
+	gateway_handlers "github.com/runatlantis/atlantis/server/lyft/gateway/events/handlers"
+	converters "github.com/runatlantis/atlantis/server/vcs/provider/github/converter"
+	"github.com/runatlantis/atlantis/server/vcs/provider/github/request"
 	"github.com/uber-go/tally"
 )
 
-const (
-	GithubHeader = "X-Github-Event"
-)
-
-type HttpResponse struct {
-	body string
-	err  HttpError
-}
-
-type HttpError struct {
-	err  error
-	code int
-}
-
-//go:generate pegomock generate -m --use-experimental-model-gen --package mocks -o mocks/mock_event_validator.go EventValidator
-type EventValidator interface {
-	InstrumentedIsValid(logger logging.SimpleLogging, baseRepo models.Repo, headRepo models.Repo, pull models.PullRequest, user models.User) bool
+const githubHeader = "X-Github-Event"
+
+func NewVCSEventsController(
+	legacyLogger logging.SimpleLogging,
+	scope tally.Scope,
+	webhookSecret []byte,
+	allowDraftPRs bool,
+	autoplanValidator gateway_handlers.EventValidator,
+	snsWriter gateway_handlers.Writer,
+	commentParser events.CommentParsing,
+	repoAllowlistChecker *events.RepoAllowlistChecker,
+	vcsClient vcs.Client,
+	logger logging.Logger,
+	supportedVCSProviders []models.VCSHostType,
+	repoConverter converters.RepoConverter,
+	pullConverter converters.PullConverter,
+) *VCSEventsController {
+	pullEventWorkerProxy := gateway_handlers.NewPullEventWorkerProxy(
+		snsWriter, logger,
+	)
+
+	asyncAutoplannerWorkerProxy := gateway_handlers.NewAsynchronousAutoplannerWorkerProxy(
+		autoplanValidator, logger, legacyLogger, pullEventWorkerProxy,
+	)
+
+	prHandler := handlers.NewPullRequestEventWithEventTypeHandlers(
+		repoAllowlistChecker,
+		asyncAutoplannerWorkerProxy,
+		asyncAutoplannerWorkerProxy,
+		pullEventWorkerProxy,
+	)
+
+	commentHandler := handlers.NewCommentEventWithCommandHandler(
+		commentParser,
+		repoAllowlistChecker,
+		vcsClient,
+		gateway_handlers.NewCommentEventWorkerProxy(logger, snsWriter),
+		logger,
+	)
+
+	// lazy map of resolver providers to their resolver
+	// laziness ensures we only instantiate the providers we support.
+	providerResolverInitializer := map[models.VCSHostType]func() events_controllers.RequestResolver{
+		models.Github: func() events_controllers.RequestResolver {
```

Comment:
So in theory, if a user appended more key-value pairs here, Atlantis can support multiple VCSHosts at once?